### PR TITLE
[#IOPID-2460] adding fn-lollipop iac

### PR DESCRIPTION
**Pull request by arcogabbo**
#### List of Changes

* [[#IOPID-2460] added fn-lollipop iac](https://github.com/pagopa/io-auth-n-identity-domain/commit/2605ba88b17161e86b125a85030d38af5c2bcf61)

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: infra/resources/prod/function_lollipop.tf [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  resource_group_name = azurerm_resource_group.function_lollipop_rg.name
+  health_check_path   = "/info"
+  node_version        = 20
+  # P2mv3 SKU and 8 Worker process count
+  tier = "xl"
+
+  subnet_cidr   = local.cidr_subnet_fn_lollipop
+  subnet_pep_id = data.azurerm_subnet.private_endpoints_subnet.id
+
+  subnet_service_endpoints = {
```
Comment (BurnedMarshal) [inline infra/resources/prod/function_lollipop.tf:83]: evaluate to remove this settings

File: infra/resources/prod/locals.tf [hunk 2]
```diff
@@ -7,9 +7,12 @@ locals {
   project            = "${local.prefix}-${local.env_short}-${local.location_short}"
   weu_project        = "${local.prefix}-${local.env_short}-${local.weu_location_short}"
   domain             = "auth"
+  full_domain        = "citizen-auth"
```
Comment (BurnedMarshal) [inline infra/resources/prod/locals.tf:10]:
```suggestion
  legacy_domain      = "citizen-auth"
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: infra/resources/prod/locals.tf (hunk 2 above)
Comment (arcogabbo) [reply infra/resources/prod/locals.tf]: done in https://github.com/pagopa/io-auth-n-identity-domain/pull/132/commits/e4d5135b95b33dec8ea56597d13eb3f56032dca0

File: infra/resources/prod/function_lollipop.tf (hunk 1 above)
Comment (arcogabbo) [reply infra/resources/prod/function_lollipop.tf]: removed in https://github.com/pagopa/io-auth-n-identity-domain/pull/132/commits/e4d5135b95b33dec8ea56597d13eb3f56032dca0

**Review by BurnedMarshal — APPROVED** [review]

**Pull request merged**